articles/implementing-firebase-in-flutter/index.md

```diff
@@ -0,0 +1,254 @@
+For simplicity, we will retrieve and print out the data in the console. This is shown below.
+
+```dart
+void printFirebase(){
+  databaseRef.once().then((DataSnapshot snapshot) {
+    print('Data : ${snapshot.value}');
+  });
+}
+```
+
+Here is the full code for the HomePage class
+
+```dart
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:firebase_database/firebase_database.dart';
+import 'package:firebase_core/firebase_core.dart';
+
+class HomePage extends StatefulWidget {
+  @override
+  _HomePageState createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  final textcontroller = TextEditingController();
+  final databaseRef = FirebaseDatabase.instance.reference();
+  final Future<FirebaseApp> _future = Firebase.initializeApp();
+
+  void addData(String data) {
+    databaseRef.push().set({'name': data, 'comment': 'A good season'});
+  }
+
+  void printFirebase(){
+    databaseRef.once().then((DataSnapshot snapshot) {
+      print('Data : ${snapshot.value}');
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    printFirebase();
+    return Scaffold(
+      appBar: AppBar(
+        title: Text("Firebase Demo"),
+      ),
+      body: FutureBuilder(
+          future: _future,
+          builder: (context, snapshot) {
+            if (snapshot.hasError) {
+              return Text(snapshot.error.toString());
+            } else {
+              return Container(
+                child: Column(
+                  children: <Widget>[
+                    SizedBox(height: 250.0),
+                    Padding(
+                      padding: EdgeInsets.all(10.0),
+                      child: TextField(
+                        controller: textcontroller,
+                      ),
+                    ),
+                    SizedBox(height: 30.0),
+                    Center(
+                        child: RaisedButton(
+                            color: Colors.pinkAccent,
+                            child: Text("Save to Database"),
+                            onPressed: () {
+                              addData(textcontroller.text);
+                              //call method flutter upload
+                            })),
+                  ],
+                ),
+              );
+            }
+          }),
+
+    );
+  }
+}
+```
+
+### 5. Results
+When we click on the Save to Database button. The user input is stored in the real-time database, as shown.
+
+![results](/engineering-educationimplementing-firebase-in-flutter/firebaseresults.png)
+
+The following items are also printed.
+
+![console results](/engineering-educationimplementing-firebase-in-flutter/console.png)
+
+### Conclusion
+We have successfully created a simple Flutter application that can store and retrieve records from the Firebase Realtime Database. You can, therefore, use this knowledge and skills to build more interactive applications.
+
```

Comment:
Generally, It will be more useful if you provide a recap of all the steps in the end for a Tutorial.
Please make note of this for your future articles and add a Recap section here instead of a conclusion.